clients/src/test/java/org/apache/kafka/clients/producer/MockProducerTest.java

```diff
@@ -77,6 +76,7 @@ public void testPartitioner() throws Exception {
         assertEquals("Partition should be correct", 1, metadata.get().partition());
         producer.clear();
         assertEquals("Clear should erase our history", 0, producer.history().size());
+        producer.close();
```

Comment:
It's odd that we close this producer, but not the one assigned to the field with the same name. Do we care about closing `MockProducer`? Did you use a tool to find cases where `Closeable`s are not being closed?